persistence/remote/src/main/java/org/infinispan/persistence/remote/RemoteStore.java

```diff
@@ -68,199 +72,244 @@
-      else
-         remoteCache = remoteCacheManager.getCache(configuration.remoteCacheName());
+      if (clusterConfiguration.cacheMode().isClustered() && !configuration.shared()) {
+         throw log.clusteredRequiresBeingShared();
+      }
+
       if (configuration.rawValues() && iceFactory == null) {
          iceFactory = ctx.getCache().getAdvancedCache().getComponentRegistry().getComponent(InternalEntryFactory.class);
       }
+
+      // Make sure threads are marked as non blocking if user didn't specify
+      configuration.properties().putIfAbsent("blocking", "false");
+
+      ConfigurationBuilder builder = buildRemoteConfiguration(configuration, marshaller);
+
+      return blockingManager.supplyBlocking(() -> {
+         remoteCacheManager = new RemoteCacheManager(builder.build());
+
+         if (configuration.remoteCacheName().isEmpty())
+            remoteCache = (InternalRemoteCache<Object, Object>) remoteCacheManager.getCache();
+         else
+            remoteCache = (InternalRemoteCache<Object, Object>) remoteCacheManager.getCache(configuration.remoteCacheName());
+
+         return remoteCache.ping();
+      }, "RemoteStore-start")
+            .thenCompose(Function.identity())
+            .thenAccept(pingResponse -> {
+               String cacheName = ctx.getCache().getName();
+               StorageConfigurationManager storageConfigurationManager = ctx.getCache().getAdvancedCache().getComponentRegistry()
+                     .getComponent(StorageConfigurationManager.class);
+               MediaType localKeyStorageType = storageConfigurationManager.getKeyStorageMediaType();
+               MediaType serverKeyStorageType = pingResponse.getKeyMediaType();
+
+               MediaType localValueStorageType = storageConfigurationManager.getValueStorageMediaType();
+               MediaType serverValueStorageType = pingResponse.getValueMediaType();
+
+               DataFormat.Builder dataFormatBuilder = DataFormat.builder().from(remoteCache.getDataFormat());
+               // TODO: I believe this is blocking...
+               Integer numSegments = remoteCache.getCacheTopologyInfo().getNumSegments();
+               boolean segmentsMatch;
+               if (numSegments == null) {
+                  log.debugf("Remote Store for cache %s cannot support segmentation as the number of segments was not found from the remote cache", cacheName);
+                  segmentsMatch = false;
+               } else {
+                  segmentsMatch = numSegments == segmentCount;
+                  if (segmentsMatch) {
+                     log.debugf("Remote Store for cache %s can support segmentation as the number of segments matched the remote cache", cacheName);
+                  } else {
+                     log.debugf("Remote Store for cache %s cannot support segmentation as the number of segments %d do not match the remote cache %d",
+                           cacheName, segmentCount, numSegments);
+                  }
```

Comment:
Do these debug statements really add much as we will throw an exception below if the configuration is not valid.

If we remove the logs we can simplify this to a one-liner:
```java
boolean segmentsMatch = numSegments == null ? false : numSegments == segmentCount;
```

Reply:
We don't always throw an exception below, so that is why I have debug statements so they can be seen.